improver/nbhood/square_kernel.py

```diff
@@ -481,13 +475,13 @@ def run(self, cube, radius, mask_cube=None):
         """
         # If the data is masked, the mask will be processed as well as the
         # original_data * mask array.
+        check_radius_against_distance(cube, radius)
         original_attributes = cube.attributes
         original_methods = cube.cell_methods
-        grid_cells_x = (
-            convert_distance_into_number_of_grid_cells(
-                cube, radius,
-                max_distance_in_grid_cells=MAX_RADIUS_IN_GRID_CELLS))
-        grid_cells_y = grid_cells_x
+        grid_cells = (
+            distance_to_number_of_grid_cells(cube, radius,
+                                             MAX_RADIUS_IN_GRID_CELLS))
```

Comment:
Oh this global variable!!!

- `MAX_RADIUS_IN_GRID_CELLS` should not be a global variable, but a local member of whatever class it's required in
- Should check whether `grid_cells` is too large in this function, not in the utility.

I will check whether we actually need this limit at all but in the meantime, would you make it a local class member please (of this or a broader neighbourhood class, if applicable).